frontend/src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,31 @@
+import {
+  Avatar as ChakraAvatar,
+  AvatarBadge as ChakraAvatarBadge,
+  AvatarProps as ChakraAvatarProps,
+} from '@chakra-ui/react'
+
+export interface AvatarProps extends ChakraAvatarProps {
+  hasNotification?: boolean
+}
+
+/*
+ * Avatar component
+ * May be wrapped in AvatarMenuButton
+ */
+export const Avatar = ({
+  hasNotification,
+  ...rest
+}: AvatarProps): JSX.Element => {
+  const avatarBadge = hasNotification ? <ChakraAvatarBadge /> : ''
+
+  return (
+    <ChakraAvatar
+      getInitials={
+        rest.getInitials ? rest.getInitials : (fullName) => fullName[0] // Extract first letter of avatar name
+      }
+      {...rest}
+    >
+      {avatarBadge}
```

Comment:
nit: should we inline this directly so it's easier to read? also prefer `null` to `''`

```tsx
<ChakraAvatar>
  {hasNotification ? <ChakraAvatarBadge /> : null}
</ChakraAvatar>
```

Reply:
thanks! edited